[Flaky Test] Root CA Controller Integration Test Suite

<!-- Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Gardener CI -->

**How to categorize this issue?**
<!--
Please select area, kind, and priority for this issue. This helps the community categorizing it.
Replace below TODOs or exchange the existing identifiers with those that fit best in your opinion.
If multiple identifiers make sense you can also state the commands multiple times, e.g.
  /area control-plane
  /area auto-scaling
  ...

"/area" identifiers:     audit-logging|auto-scaling|backup|certification|control-plane-migration|control-plane|cost|delivery|dev-productivity|disaster-recovery|documentation|high-availability|logging|metering|monitoring|networking|open-source|ops-productivity|os|performance|quality|robustness|scalability|security|storage|testing|usability|user-management
"/kind" identifiers:     api-change|bug|cleanup|discussion|enhancement|epic|impediment|poc|post-mortem|question|regression|task|technical-debt|test
-->
/area testing
/kind flake

**Which test(s)/suite(s) are flaking**:
`Root CA Controller Integration Test Suite`

**CI link**: https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-master/jobs/master-head-update-job/builds/185

**Reason for failure**:

```
Running Suite: Root CA Controller Integration Test Suite
========================================================
Random Seed: 1635963430
Will run 5 of 5 specs

•
------------------------------
• Failure in Spec Teardown (AfterEach) [1.005 seconds]
RootCAPublisher tests
/go/src/github.com/gardener/gardener/test/integration/envtest/resourcemanager/rootcapublisher/publisher_test.go:27
  kube-root-ca.crt config map
  /go/src/github.com/gardener/gardener/test/integration/envtest/resourcemanager/rootcapublisher/publisher_test.go:48
    controller should be active
    /go/src/github.com/gardener/gardener/test/integration/envtest/resourcemanager/rootcapublisher/publisher_test.go:55
      should revert the config map data after manual changes [AfterEach]
      /go/src/github.com/gardener/gardener/test/integration/envtest/resourcemanager/rootcapublisher/publisher_test.go:65

      Timed out after 1.001s.
      Expected
          <map[string]string | len:0>: nil
      to have {key: value}
          <map[interface {}]interface {} | len:1>: {
              <string>"ca.crt": <string>"-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----\n",
          }

      /go/src/github.com/gardener/gardener/test/integration/envtest/resourcemanager/rootcapublisher/publisher_test.go:60
------------------------------
•••

Summarizing 1 Failure:

[Fail] RootCAPublisher tests kube-root-ca.crt config map controller should be active [AfterEach] should revert the config map data after manual changes 
/go/src/github.com/gardener/gardener/test/integration/envtest/resourcemanager/rootcapublisher/publisher_test.go:60

Ran 5 of 5 Specs in 8.746 seconds
FAIL! -- 4 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestRootCAPublisher (8.75s)
FAIL
```

--- a/hack/test-integration.sh
+++ b/hack/test-integration.sh
@@ -33,4 +33,7 @@ echo "using envtest tools installed at '$KUBEBUILDER_ASSETS'"
 echo "> Integration Tests"
 
 export KUBEBUILDER_CONTROLPLANE_START_TIMEOUT=2m
+export GOMEGA_DEFAULT_EVENTUALLY_TIMEOUT=5s
+export GOMEGA_DEFAULT_EVENTUALLY_POLLING_INTERVAL=200ms
+
 GO111MODULE=on go test -timeout=5m -mod=vendor $@ | grep -v 'no test files'

--- a/test/integration/envtest/resourcemanager/tokenrequestor/tokenrequestor_test.go
+++ b/test/integration/envtest/resourcemanager/tokenrequestor/tokenrequestor_test.go
@@ -19,13 +19,11 @@ import (
 
 	"github.com/gardener/gardener/pkg/client/kubernetes"
 	. "github.com/gardener/gardener/pkg/utils/test/matchers"
-
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
-	kubernetesscheme "k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
 	clientcmdlatest "k8s.io/client-go/tools/clientcmd/api/latest"
 	clientcmdv1 "k8s.io/client-go/tools/clientcmd/api/v1"
@@ -123,7 +121,7 @@ var _ = Describe("TokenRequestor tests", func() {
 
 		AfterEach(func() {
 			var newClient client.Client
-			newClient, err = client.New(newRestConfig, client.Options{Scheme: kubernetesscheme.Scheme})
+			newClient, err = client.New(newRestConfig, client.Options{Mapper: testClient.RESTMapper()})
 			Expect(err).ToNot(HaveOccurred())
 
 			Eventually(func() error {

--- a/test/integration/envtest/seedadmissioncontroller/extensioncrds_test.go
+++ b/test/integration/envtest/seedadmissioncontroller/extensioncrds_test.go
@@ -17,7 +17,6 @@ package seedadmissioncontroller_test
 import (
 	"context"
 	"fmt"
-	"time"
 
 	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
@@ -115,7 +114,7 @@ var _ = Describe("Extension CRDs Webhook Handler", func() {
 				}
 			}
 			return true
-		}, 5*time.Second, 200*time.Millisecond).Should(BeTrue())
+		}).Should(BeTrue())
 	})
 
 	objectID := func(obj client.Object) string {
@@ -126,34 +125,34 @@ var _ = Describe("Extension CRDs Webhook Handler", func() {
 		Eventually(func() string {
 			err := c.Delete(ctx, obj)
 			return string(apierrors.ReasonForError(err))
-		}, 5*time.Second, 200*time.Millisecond).Should(ContainSubstring("annotation to delete"), objectID(obj))
+		}).Should(ContainSubstring("annotation to delete"), objectID(obj))
 	}
 
 	testDeleteCollectionUnconfirmed := func(ctx context.Context, obj client.Object) {
 		Eventually(func() string {
 			err := c.DeleteAllOf(ctx, obj, client.InNamespace(obj.GetNamespace()))
 			return string(apierrors.ReasonForError(err))
-		}, 5*time.Second, 200*time.Millisecond).Should(ContainSubstring("annotation to delete"), objectID(obj))
+		}).Should(ContainSubstring("annotation to delete"), objectID(obj))
 	}
 
 	testDeletionConfirmed := func(ctx context.Context, obj client.Object) {
 		Eventually(func() error {
 			return c.Delete(ctx, obj)
-		}, 5*time.Second, 200*time.Millisecond).ShouldNot(HaveOccurred(), objectID(obj))
+		}).ShouldNot(HaveOccurred(), objectID(obj))
 		Eventually(func() bool {
 			err := c.Get(ctx, client.ObjectKeyFromObject(obj), obj)
 			return apierrors.IsNotFound(err) || meta.IsNoMatchError(err)
-		}, 5*time.Second, 200*time.Millisecond).Should(BeTrue(), objectID(obj))
+		}).Should(BeTrue(), objectID(obj))
 	}
 
 	testDeleteCollectionConfirmed := func(ctx context.Context, obj client.Object) {
 		Eventually(func() error {
 			return c.DeleteAllOf(ctx, obj, client.InNamespace(obj.GetNamespace()))
-		}, 5*time.Second, 200*time.Millisecond).ShouldNot(HaveOccurred(), objectID(obj))
+		}).ShouldNot(HaveOccurred(), objectID(obj))
 		Eventually(func() bool {
 			err := c.Get(ctx, client.ObjectKeyFromObject(obj), obj)
 			return apierrors.IsNotFound(err) || meta.IsNoMatchError(err)
-		}, 5*time.Second, 200*time.Millisecond).Should(BeTrue(), objectID(obj))
+		}).Should(BeTrue(), objectID(obj))
 	}
 
 	Context("extension resources", func() {

